File: 01-FUNDAMENTOS_DE_PROGRAMACAO/Dart/exercicio2.dart
// 2 - Dado um número inteiro positivo de 3 algarismos, inverter a ordem de seus
// algarismos. Os três algarismos do número dado são diferentes de zero.

import 'dart:io';

main(List<String> args) {
  print('Digite um número inteiro de 3 algarismos diferente de 0: ');
  var number = int.parse(stdin.readLineSync()!);

  int centena = number ~/ 100;
  int dezena = (number % 100) ~/ 10;
  int unidade = (number % 100) % 10;
  print("O número invertido é ${unidade}${dezena}${centena}");
}

File: 01-FUNDAMENTOS_DE_PROGRAMACAO/TypeScript/Exercicio2.ts
/*
2 - Dado um número inteiro positivo de 3 algarismos, inverter a ordem de seus
algarismos. Os três algarismos do número dado são diferentes de zero.
*/

//1ª Forma
const num: number = 125
let centena = (num / 100) | 0
let dezena = ((num % 100) / 10) | 0
let unidade = (num % 10) % 10
console.log(unidade, dezena, centena)

//2ª Forma
const num3: number = 125
centena = Math.floor(num3 / 100)
dezena = Math.floor((num3 % 100) / 10)
unidade = (num3 % 10) % 10
console.log(unidade, dezena, centena)

//3ª Forma
const outro: string = '125'
console.log(outro.charAt(2), outro.charAt(1), outro.charAt(0))


